Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp

```diff
@@ -271,6 +272,17 @@ void VideoFrame::paintInContext(GraphicsContext& context, const FloatRect& desti
     context.drawImage(image->image(), destination, source, { compositeOperator, destinationImageOrientation });
 }
 
+uint32_t VideoFrameGStreamer::pixelFormat() const
+{
+    if (m_cachedVideoFormat != GST_VIDEO_FORMAT_UNKNOWN)
+        return m_cachedVideoFormat;
```

Comment:
I wonder if the video format could change if the caps change, for example, rendering this cache incorrect at that point. Maybe invalidating the cache at some other point?

Reply:
a video frame wraps a sample basically, so I don't see in which case the sample caps would change.